lib/libshare/os/linux/nfs.c

```diff
@@ -407,77 +405,124 @@ get_linux_shareopts(const char *shareopts, char **plinux_opts)
 /*
  * Enables NFS sharing for the specified share.
  */
-static int
-nfs_enable_share(sa_share_impl_t impl_share)
+static char *
+nfs_init_tmpfile(void)
 {
-	char *shareopts, *linux_opts;
-	int rc;
+	char *tmpfile = NULL;
 
-	if (!nfs_available()) {
+	if (asprintf(&tmpfile, "%s%s", ZFS_EXPORTS_FILE, ".XXXXXX") == -1) {
+		fprintf(stderr, "Unable to allocate temporary file\n");
+		return (NULL);
+	}
```

Comment:
Since you aren't using it here (which I think is fine), I think you can get rid of ZFS_EXPORTS_FILE_TEMP.